ifupdown2/ifupdown/networkinterfaces.py

```diff
@@ -176,6 +181,25 @@ def process_source(self, lines, cur_idx, lineno):
                     'unable to read source line')
         return 0
 
+    def process_source_directory(self, lines, cur_idx, lineno):
+        self.logger.debug('processing source-directory line ..\'%s\'' % lines[cur_idx])
+        sourced_directory = re.split(self._ws_split_regex, lines[cur_idx], 2)[1]
+
+        if sourced_directory:
+            if not os.path.isabs(sourced_directory):
+                sourced_directory = os.path.join(os.path.dirname(self._currentfile), sourced_directory)
+
+            folders = glob.glob(sourced_directory)
+            for folder in folders:
+                filenames = [file for file in os.listdir(folder) if re.match(r'^[a-zA-Z0-9_-]+$', file)]
```

Comment:
Why do we need a name check here? 
The following doesn't work with this regex check:
```
➜  ~ cat /etc/network/interfaces             
auto eth0
iface eth0 inet dhcp

source-directory interfaces.d
➜  ~ ll /etc/network/interfaces.d/
total 4.0K
-rw-r--r-- 1 root root 45 Dec 17 03:19 test2.intf
➜  ~ cat /etc/network/interfaces.d/test2.intf
auto test2
iface test2
      link-type dummy
➜  ~ 
➜  ~ ifquery -a
auto eth0
iface eth0 inet dhcp

➜  ~ 
```

Wouldn't:
```
+            for folder in glob.glob(sourced_directory):
+                for f in os.listdir(folder) or []:
+                    self.read_file(os.path.join(folder, f))
```
work just as well?

Reply:
That is expected to match ifupdown's behaviour, see https://salsa.debian.org/debian/ifupdown/-/blob/0.8.36/interfaces.5.pre#L119-125

Reply:
I have nothing to add to @sbraz's explanation. That was exactly my intention.

Reply:
I think we can extend ifupdown's behavior and drop the regex check to cover a broader range of file names (like the one I used: ".intf").

What do you guys think?